updateHostsFile.py

```diff
@@ -1126,6 +1129,60 @@ def remove_old_hosts_file(backup):
 # End File Logic
 
 
+def domain_to_idna(line):
+    """
+    Encode a domain which is presente into a line into `idna`. This way we
+    avoid the most encoding issue.
+
+    Parameters
+    ----------
+    line : str
+        The line we have to encode/decode.
+
+    Returns
+    -------
+    line : str
+        The line in a converted format.
+
+    Notes
+    -----
+    - This method/function encode only the domain to `idna` format because in
+        most cases the encoding issue is due to a domain which looks like
+        `b'\xc9\xa2oogle.com'.decode('idna')`.
+    - About the splitting:
+        We split because we only want to encode the domain and not the full
+            line which may cause some issue. Keep in mind that we split but we
+            still concatenate once we encoded the domain.
```

Comment:
Tabbing looks weird here.

Reply:
Grammar nits:

"line, which may cause some issues" (add comma + pluralize issue)
"split, but we still" (add comma after 'split')

Reply:
Fixed :white_check_mark: